### DISCUSSION TITLE
Assembler Intellisense and Macro expansion

### DISCUSSION THREAD

**Discussion by spufidoo** [IBM/zopeneditor-about#167] [Q&A]
I don't know whether this is an "issue" or just my ignorance, but I'm having difficulty in getting zopeneditor to expand macros.
I hover over a macro name **$USING** in my code, and I get "**Operation code not found - $USING** HLASM Plugin(E049)"
My file structure looks like this:
c:\project\asm
c:\project\maclib
I know there should be a JSON entry in some settings file somewhere, but I can't find it.
Any clues, anyone?

**Comment by phaumer** [comment]
Hello @spufidoo. We currently support only two ways of including files:

1. via COPY statements using [property groups](https://ibm.github.io/zopeneditor-about/Docs/editor_codechanges_hlasm.html#previewing-copybooks)
2. macros that are defined in `SYS1.MACLIB` [loaded automatically if switched on](https://ibm.github.io/zopeneditor-about/Docs/editor_codechanges_hlasm.html#remote-macros) requiring a Zowe CLI profile 

We currently do not have support for custom macros, yet. If you are interested in that please open an issue with some more details how you want to use them (locally or on MVS, or both).

**Comment by FALLAI-Denis** [comment]
Relate to #124 [New extension to link to external Documentary Base (as IBM Docs)](https://github.com/IBM/zopeneditor-about/issues/124) ?
